shared/chat/conversation/messages/react-button/emoji-picker/container.tsx

```diff
@@ -52,100 +53,147 @@ const useReacji = ({onPick, onDidPick}: Props) => {
-        containerStyle={styles.input}
-        decoration={
-          <Kb.Text type="BodySemiboldLink" onClick={onCancel}>
-            Cancel
-          </Kb.Text>
-        }
-        placeholder="Search"
-        icon="iconfont-search"
-        onChangeText={filter => setFilter(filter)}
-        textType="BodySemibold"
-      />
+    <Kb.Box2 direction="vertical" onLayout={onLayout} fullWidth={true} fullHeight={true}>
+      <Kb.Box2 direction="horizontal" fullWidth={true} alignItems="center">
+        <Kb.ClickableBox onClick={onCancel} style={styles.cancelContainerMobile}>
+          <Kb.Text type="BodyBigLink">Cancel</Kb.Text>
+        </Kb.ClickableBox>
+        <Kb.SearchFilter
+          focusOnMount={true}
+          size="small"
+          icon="iconfont-search"
+          placeholderText="Search"
+          onChange={str => setFilter(str)}
+          style={styles.searchFilter}
+        />
+      </Kb.Box2>
       <EmojiPicker
         topReacjis={topReacjis}
         filter={filter}
-        onChoose={emoji => onAddReaction(`:${emoji.short_name}:`)}
+        onChoose={emoji => onAddReaction(emoji)}
         width={width}
+        skinTone={currentSkinTone}
       />
+      <Kb.Box2 direction="horizontal" fullWidth={true} alignItems="center" style={styles.footerContainer}>
+        <SkinTonePicker currentSkinTone={currentSkinTone} setSkinTone={setSkinTone} />
+      </Kb.Box2>
     </Kb.Box2>
   )
 }
 
 export const EmojiPickerDesktop = (props: Props) => {
   const {filter, onAddReaction, setFilter, topReacjis} = useReacji(props)
+  const {currentSkinTone, setSkinTone} = useSkinTone()
   return (
-    <Kb.Box
-      direction="vertical"
-      style={styles.containerDesktop}
-      onClick={e => e.stopPropagation()}
-      gap="tiny"
-    >
-      <Kb.SearchFilter
-        focusOnMount={true}
-        size="full-width"
-        icon="iconfont-search"
-        placeholderText="Search"
-        onChange={str => setFilter(str)}
-        style={styles.searchFilter}
-      />
+    <Kb.Box style={styles.containerDesktop} onClick={e => e.stopPropagation()} gap="tiny">
+      <Kb.Box2
+        direction="horizontal"
+        gap="tiny"
+        fullWidth={true}
+        alignItems="center"
+        style={styles.topContainerDesktop}
+      >
+        <Kb.SearchFilter
+          focusOnMount={true}
+          size="full-width"
+          icon="iconfont-search"
+          placeholderText="Search"
+          onChange={str => setFilter(str)}
+        />
+        <SkinTonePicker currentSkinTone={currentSkinTone} setSkinTone={setSkinTone} />
+      </Kb.Box2>
       <Kb.Box style={styles.emojiContainer}>
         <EmojiPicker
           topReacjis={topReacjis}
           filter={filter}
-          onChoose={emoji => onAddReaction(`:${emoji.short_name}:`)}
-          width={320}
+          onChoose={emoji => onAddReaction(emoji)}
+          width={336}
+          skinTone={currentSkinTone}
         />
       </Kb.Box>
+      {/* TODO
```

Comment:
?

Reply:
This is the footer on desktop. I'll do that in a different ticket.